shell/pages/c/_cluster/settings/performance.vue

```diff
@@ -89,8 +101,39 @@ export default {
     </h1>
     <div>
       <div class="ui-perf-setting">
-        <!-- Websocket Notifications -->
         <div class="mt-20">
+          <h2>{{ t('inactivity.settings.title') }}</h2>
+          <p>{{ t('inactivity.settings.description') }}</p>
+          <Checkbox
+            v-model="value.inactivity.enabled"
+            :mode="mode"
+            :label="t('inactivity.settings.checkboxLabel')"
+            class="mt-10 mb-20"
+            :primary="true"
+          />
+          <div class="ml-20">
+            <LabeledInput
+              v-model="value.inactivity.threshold"
+              :mode="mode"
+              :label="t('inactivity.settings.inputLabel')"
+              :disabled="!value.inactivity.enabled"
+              class="input mb-10"
+              type="number"
+              min="0"
+            />
+            <Banner
```

Comment:
This should probably use the existing validation feature. There should be a prop that accepts a fn which if it fails will return an error message. There should be some docs of it in our dev kit somewhere, though if in doubt reach out to Sean

Reply:
That might also help with the visually valid `-1` (don't think `LabeledInput` emits a value for minus figures but doesn't block the user from entering it)

Reply:
We had a really wide range of validation cases in https://github.com/rancher/dashboard/issues/6633 which may be worthy to reuse if the requirements are the same.
Specifically [these](https://github.com/rancher/dashboard/blob/master/shell/config/settings.ts#L102-L113) rules:
>
```
      {
        name: 'isInteger',
        key:  'Password', <-- term to be replaced for the error
      },
      {
        name: 'isPositive',
        key:  'Password',
      },
      {
        name: 'isOctal',
        key:  'Password',
      }
```

Reply:
For the records the prop is `rules` and the method of the mixin is `fvGetAndReportPathRules`.
It would be required to add the whole validation to the CRUD component in order to have a real meaning.

Some info about it are defined in the devkit [here](https://rancher.github.io/dashboard/code-base-works/forms-and-validation#form-validation).

Reply:
@richard-cox that page is full of inputs which have no validation at all.
I would recommend to create a validation for the whole page in a separated issue, otherwise we'll have issues with the consistency.

Reply:
This was a small task, and to avoid introducing technical debt and a different validation ux i've pushed a commit with the fix in